client/my-sites/email/titan-add-mailboxes/titan-new-mailbox-list.jsx

```diff
@@ -27,6 +27,7 @@ const TitanNewMailboxList = ( {
 	onMailboxesChange,
 	onReturnKeyPress = noop,
 	showLabels = true,
+	supportsMultipleMailboxes = true,
```

Comment:
🤔  I feel `supportsMultipleMailboxes` depicts this better. 

In computing we often say a feature is **supported** if that feature is accessible/capable to be used/managed. 

Most times if a feature is **allowed** , it may be related to permission/authorisation. Better put, a feature may be supported, but not allowed.

Reply:
Sorry about not explaining my reasoning -- most of my concern was that `supportsMultipleMailboxes` is somewhat passive and is more of a state than an instruction/command for the component. From that perspective, I interpret/understand `allowMultipleMailboxes` to be an instruction for the component to affect/direct its behaviour, much like the `showLabels` prop.
I could see `supportMultipleMailboxes` being a valid name using that naming approach, because we're giving the component an instruction.
But I don't think this naming conversation should block the PR! I'll kick off a dedicated async discussion on this.

Reply:
I've already mentioned it internally but I don't have a strong opinion about these two options (I would use `supportMultipleMailboxes` instead of `supportsMultipleMailboxes` though). `showAddAnotherMailboxButton` could be a good alternative that better conveys what that flag actually does.